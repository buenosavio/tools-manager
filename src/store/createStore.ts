/* eslint-disable import/no-anonymous-default-export */
import { createStore, applyMiddleware, Reducer, Store } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { routerMiddleware } from 'connected-react-router';
import { SagaMiddleware } from 'redux-saga';
import { History } from 'history';

const isDevelopMode = process.env.NODE_ENV === 'development';

export default (reducers: Reducer, sagaMiddleware: SagaMiddleware, history: History): Store => {
  if (isDevelopMode) {
    const storeEnhacer = composeWithDevTools({ trace: true, traceLimit: 30 })(
      applyMiddleware(sagaMiddleware, routerMiddleware(history))
    );
    return createStore(reducers, storeEnhacer);
  }
  return createStore(reducers, applyMiddleware(sagaMiddleware, routerMiddleware(history)));
};
