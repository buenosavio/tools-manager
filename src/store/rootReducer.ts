import ToolsReducer from './tools/duck'
import {combineReducers} from '@reduxjs/toolkit'

const rootReducer = combineReducers({
  ToolsReducer
})


export type AppState = ReturnType<typeof rootReducer>

export default rootReducer