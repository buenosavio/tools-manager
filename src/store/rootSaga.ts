import {all} from 'redux-saga/effects'
import toolSaga from './tools/saga'

export default function* rootSaga(): Generator {
  yield all([toolSaga()])
}