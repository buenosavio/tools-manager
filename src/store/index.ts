import createSagaMiddleware from "redux-saga";
import createStore from "./createStore";
import rootReducer from "./rootReducer";
import history from "./history";
import { persistReducer } from "redux-persist";
import persistStore from "redux-persist/es/persistStore";
import rootSaga from "./rootSaga";
import storage from 'redux-persist/lib/storage'

const sagaMiddleware = createSagaMiddleware()

const store = createStore(persistReducer({key: 'store', storage },rootReducer), sagaMiddleware, history);

const persistor = persistStore(store);

sagaMiddleware.run(rootSaga)

export type AppDispatch = typeof store.dispatch;

export {store, persistor}