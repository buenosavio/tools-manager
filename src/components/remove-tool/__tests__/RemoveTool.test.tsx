import renderer from 'react-test-renderer'
import { store } from '../../../store'
import withProvider from '../../../utils/withContext'
import RemoveTool from '../RemoveTool'

describe('RemoveTool', () => {
  const item = { tools: [], isOnlyTag: true, loading: true, search: '' }
  const Component = withProvider(RemoveTool, store)
  test('Snapshot of component', () => {
    const tree = renderer
      .create(<Component item={item} handleOpen={() => {}} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
