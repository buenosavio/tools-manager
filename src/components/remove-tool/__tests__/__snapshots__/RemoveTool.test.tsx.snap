// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveTool Snapshot of component 1`] = `
<div
  className="sc-bczRLJ gZNRzs"
>
  <div
    className="sc-gsnTZi bQzYaW"
  >
    <div
      className="sc-dkzDqf ciSiFa"
    >
      <img
        alt="Imagem de um Ã­cone em X"
        src="removeIcon.svg"
      />
      Remove tool
    </div>
    <h3>
      Are sure you want to remove?
    </h3>
    <footer
      className="sc-iqcoie efPTbX"
    >
      <button
        className="sc-jqUVSM TvhFC"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="sc-kDDrLX hUoWzT"
        onClick={[Function]}
      >
        Yes, remove
      </button>
    </footer>
  </div>
</div>
`;
